modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/consensus/block/BlockConsensusCell.scala

```diff
@@ -340,15 +339,27 @@ object BlockConsensusCell {
               if gotAllBlockProposals(roundData) =>
             for {
               _ <- Applicative[F].unit
-              finalBlock = roundData.peerBlocks.values.fold(ownBlock)(_ |+| _)
-              result <- finalBlock.toHashedWithSignatureCheck.flatMap {
-                case Left(_) =>
-                  cancelRound(roundData.ownProposal, ctx)
-                    .map(_ => CellError("Round cancelled after final block turned out to be invalid!").asLeft[Ω])
-
-                case Right(hashedBlock) =>
-                  cleanUpRoundData(roundData.ownProposal, ctx)
-                    .map(_ => FinalBlock(hashedBlock).asRight[CellError].widen[Ω])
+              maybeFinalBlock = roundData.peerBlocks.values.foldLeft {
+                ownBlock.asRight[MergingProofsForDistinctValues[DAGBlock]]
+              } { case (agg, sb) => agg.flatMap(_.addProofs(sb)) }
+              result <- maybeFinalBlock match {
+                case Left(error) =>
+                  cancelRound(roundData.ownProposal, ctx).map(
+                    _ =>
+                      CellError(
+                        s"Round cancelled after merging blocks to a final block failed! Original message: ${error.getMessage}"
```

Comment:
So "Round cancelled" or "Round ... failed" :D

Reply:
But the round got cancelled as a result of a failure during blocks merging ;).